### PR TITLE
Restrict to only 1 obs/year: animals & herbaceous

### PR DESCRIPTION
**Pull request by ya-web**

### CODE REVIEW

**Review by kstefanini — CHANGES_REQUESTED** [review]

File: src/Service/YearsIndividualObservations.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            $year = date_format($observation->getDate(), 'Y');
+            $i = array_search($year, $obsYears);
+            if (false === $i) {
+                $obsYears[] = $year;
+                $observationsPerYear[] = [
+                    'year' => $year,
+                    'observations' => [$observation],
+                ];
+            } else {
+                $observationsPerYear[$i]['observations'][] = $observation;
```
[the remark this answers is not in the document]
Comment (ya-web) [reply src/Service/YearsIndividualObservations.php:57]: $i is the index the array structure is like
[
  [
    'year' => 2019,
    'observations' => [ obs1,obs2,...]
  ]
  [
    'year' => 2020,
    'observations => [...]
  ]
];

File: src/Service/YearsIndividualObservations.php [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        $observationsPerYear = [];
+        $obsYears = [];
+
+        /**
+         * @var Observation $observation
+         */
+        foreach ($this->validIndividualObservations as $observation) {
+            $year = date_format($observation->getDate(), 'Y');
+            $i = array_search($year, $obsYears);
+            if (false === $i) {
```
[the remark this answers is not in the document]
Comment (ya-web) [reply src/Service/YearsIndividualObservations.php:50]: it won't work see this array's structure below

File: src/Service/YearsIndividualObservations.php [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                $obsYears[] = $year;
+                $observationsPerYear[] = [
+                    'year' => $year,
```
[the remark this answers is not in the document]
Comment (ya-web) [reply src/Service/YearsIndividualObservations.php:53]: $observationsPerYear is a non associative array of associative arrays see aray's structure below

File: src/Service/YearsIndividualObservations.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (ya-web) [reply src/Service/YearsIndividualObservations.php:57]: no $i is an index see array's structure below

File: src/Service/YearsIndividualObservations.php [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    {
+        $observationsPerYear = [];
+        $obsYears = [];
+
+        /**
+         * @var Observation $observation
+         */
+        foreach ($this->validIndividualObservations as $observation) {
+            $year = date_format($observation->getDate(), 'Y');
+            $i = array_search($year, $obsYears);
```
[the remark this answers is not in the document]
Comment (ya-web) [reply src/Service/YearsIndividualObservations.php:49]: yes, it is the position of the array containing the year and the occurences

File: src/Service/YearsIndividualObservations.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (ya-web) [reply src/Service/YearsIndividualObservations.php:50]:
```
[
  [
    'year' => 2019,
    'observations' => [ obs1,obs2,...]
  ]
  [
    'year' => 2020,
    'observations => [...]
  ]
];
```

File: src/Service/YearsIndividualObservations.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (ya-web) [reply src/Service/YearsIndividualObservations.php:53]:
```
[
  [
    'year' => 2019,
    'observations' => [ obs1,obs2,...]
  ]
  [
    'year' => 2020,
    'observations => [...]
  ]
];
```

**Pull request closed without merging**